packages/flutter/lib/src/rendering/image.dart

```diff
@@ -161,8 +175,9 @@ class RenderImage extends RenderBox {
     if (constraints.isTight || _image == null)
       return constraints.smallest;
 
-    double width = _image.width.toDouble();
-    double height = _image.height.toDouble();
+    double scale = _scale ?? 1.0;
```

Comment:
width and height "null" mean "determine the value automatically from the pixels". I'm all for having a way to do that with the scale, but that's not what the code here does, it just defaults to 1.0 regardless of the image data. :-)